codec-http/src/main/java/io/netty/handler/codec/http/HttpServerKeepAliveHandler.java

```diff
@@ -82,6 +82,7 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
             }
         }
         if (msg instanceof LastHttpContent && !shouldKeepAlive()) {
+            promise = promise.unvoid();
             promise.addListener(ChannelFutureListener.CLOSE);
```

Comment:
Or you mean this 

`promise = promise.unvoid().addListener(ChannelFutureListener.CLOSE);`

?